### PR TITLE
Adding HTTP component unit tests

### PR DESCRIPTION
**Pull request by divekarshubham**
*Issue #, if available:*

*Description of changes:* Adding unit tests for http

Created a separate folder with ota_config.h to set data over http and perform basic http component initialization and use test cases.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: test/unit-test-http/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,88 @@
+# Include filepaths for source and include.
```
Comment (tgsong) [inline test/unit-test-http/CMakeLists.txt:1]: We already have one under `test/unit-test/CMakeLists.txt`, we duplicating this file?

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -85,13 +85,14 @@ include( ${MODULE_ROOT_DIR}/tools/cmock/create_test.cmake )
 
 # Include build configuration for unit tests.
 add_subdirectory( unit-test )
+add_subdirectory( unit-test-http )
```
Comment (tgsong) [inline test/CMakeLists.txt:88]: Why not just put it under `unit-test`?

**Review by tgsong — CHANGES_REQUESTED** [review]
There are lots of duplicated code, only real test is less than 10 lines. Let's refactor and merge with existing unit tests.

File: test/CMakeLists.txt (hunk 2 above)
Comment (divekarshubham) [reply test/CMakeLists.txt:88]: For data over HTTP we need to set the configuration in ota_config.h
We would need a separate folder to keep this file.

File: test/unit-test-http/CMakeLists.txt (hunk 1 above)
Comment (divekarshubham) [reply test/unit-test-http/CMakeLists.txt]: Agreed, changed structure in https://github.com/aws/ota-for-aws-iot-embedded-sdk/pull/62/commits/8f2a5031f8d914cbd8690c3cfe4c3c3ecc817782

**Comment by tgsong** [comment]
Close for now, to be merged with ota agent unit tests.

**Pull request closed without merging**